### ISSUE TITLE
Usage of deprecated Lambda runtime (Node.js 12)

### ISSUE THREAD

**Issue by tvancasteren** [awslabs/aws-dataall#425]
### Describe the bug

Data.all makes use of a number of Lambda functions leveraging the runtime Node.js 12. This Lambda runtime is deprecated and not receiving security updates anymore, resulting in security alerts from AWS Trusted Advisor.

This involves the following Lambda functions:

Deployment account (us-east-1 region, 8 Lambdas affected):
dataall-dev-cloudfront-st-UserPoolClientUpdateHand-gPbr3IcD1BGK
dataall-dev-cloudfront-stage-clou-ParseAuthHandler-Rdf18PSzydcP
dataall-dev-cloudfront-stage-clou-CheckAuthHandler-ubksBahEPw7E
dataall-dev-cloudfront-sta-LambdaCodeUpdateHandler-BnhSbQEF7Cej
dataall-dev-cloudfront-stage--RandomValueGenerator-Ji35iJprMkBa
dataall-dev-cloudfront-st-UserPoolDomainLookupHand-FVicTspGgtBK
dataall-dev-cloudfront-stage-cl-RefreshAuthHandler-aoiGg3TCAbU3
dataall-dev-cloudfront-stage-cloudf-SignOutHandler-cdgdUqRryYZB

Deployment account (home region, 3 Lambdas affected):
dataall-dev-backend-stage-CognitoProviderdevframew-3Jf9ZUyKuHjE
dataall-dev-backend-stage-AWS679f53fac002430cb0da5-nVWUrhAoUikr
dataall-dev-backend-stage-CustomS3AutoDeleteObject-7sp1Sa8kKoTF

Tooling account (home region, 1 Lambda affected):
dataall-main-cicd-stack-CustomS3AutoDeleteObjectsC-nABrKFVwz4c8

Each account onboarded as an environment (region of choice, 1 Lambda affected):
dataall-environment-4kkia-dataallLakeformationDefa-4D9ipp0vGxhX

Lambda runtimes doc: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

![image](https://user-images.githubusercontent.com/105369345/233339595-116e60c8-b149-48af-b7cc-1bb336245aab.png)

### How to Reproduce

A default deployment of data.all will result in these alerts.

### Expected behavior

Update of all affected data.all Lambda functions to a supported runtime version.

### OS

Node.js 12

### Python version

Node.js 12

### AWS data.all version

v1.4.3

**Comment by dlpzx** [comment]
I can confirm that V1.5 still has this issue, thanks for raising it we are looking into it

**Comment by dlpzx** [comment]
The Lambdas with deprecated runtimes are those created automatically by cdk built-in methods. For example, `dataall-main-cicd-stack-CustomS3AutoDeleteObjectsC-nABrKFVwz4c8` is created whenever `removal_policy=RemovalPolicy.DESTROY` is set on an S3 Bucket. 

For the tooling account:
``` in deploy/stacks/pipeline.py
self.pipeline_bucket = s3.Bucket(
            self,
            id='source-code-bucket',
            bucket_name=self.pipeline_bucket_name,
            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
            removal_policy=RemovalPolicy.DESTROY,
```

We use `aws-cdk-lib==2.61.1`. I am checking what is the default runtime used in this version.

**Comment by dlpzx** [comment]
@tvancasteren if your deployment of data.all is in version V1.4 or lower, the issue can be solved by upgrading to V1.5. I verified that deployments that were on lower versions update the node version once they upgraded to V1.5.


Alternatively you could also upgrade your version of `cdk-lib`, but we recommend you to upgrade to V1.5 to keep up with the latest changes and fixes.

**Comment by tvancasteren** [comment]
Thanks @dlpzx for the alignment we've had on this.
